src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -810,11 +810,16 @@ public EmbeddedTrimmingDescriptorNode EmbeddedTrimmingDescriptor(EcmaModule modu
             return _embeddedTrimmingDescriptors.GetOrAdd(module);
         }
 
-        private NodeCache<GCPointerMap, GCStaticEETypeNode> _GCStaticEETypes;
+        private NodeCache<(GCPointerMap, bool), GCStaticEETypeNode> _GCStaticEETypes;
 
-        public ISymbolNode GCStaticEEType(GCPointerMap gcMap)
+        public ISymbolNode GCStaticEEType(GCPointerMap gcMap, bool requiredAlign8)
         {
-            return _GCStaticEETypes.GetOrAdd(gcMap);
+            // Ignore requiredAlign8 flag for anything but ARM32 and WASM
+            if (Target.Architecture is not TargetArchitecture.ARM and not TargetArchitecture.Wasm32)
+            {
+                requiredAlign8 = false;
+            }
```

Comment:
It would be nice for this hardcoded architecture check to be lifted into a property on `Target` (e. g. `RequiresAlign8`). There are other places that hardcode "Align8 <=> ARM32" that could be changed to use it (and that had to be modified downstream).